test/test_lrf.cpp

```diff
@@ -76,21 +76,34 @@ TEST(TestLRF, renderLineBack)
 TEST(TestLRF, renderLineWeird)
 {
     geo::LaserRangeFinder lrf;
-    lrf.setAngleLimits(-M_PI_2, M_PI_2); // angle limits +- 90 degrees
     lrf.setNumBeams(30);
     lrf.setRangeLimits(RANGE_MIN, RANGE_MAX);
 
-    geo::Vec2d p1(1.0, 1.5); // point in view of the robot
-    geo::Vec2d p2(-2.0, -0.75); // in the blind spot of the robot at a negative angle ~ -2.78 RAD
-    // the line connecting these two points passes behind the robot. But the difference in angle in view of the robot is less than 180%
+    double a_max = M_PI_2;
+    double a1 = a_max - 0.1; // point in view of the robot -> -a_max < a1 < a_max
+    double a2 = -a_max - 1.0; // in the blind spot of the robot but with a positive angle -> a2 > 0
+    // the line connecting these two points should pass behind the robot -> a1 - a2 > M_PI
+    // However the difference between the a1 and the lower angle limit is less than half a circle -> a1 + a_max < M_PI
+
+    lrf.setAngleLimits(-a_max, a_max);
+
+    geo::Vec2d p1(cos(a1), sin(a1)); 
+    geo::Vec2d p2(cos(a2), sin(a2)); 
     std::vector<double> ranges(N_BEAMS, RANGE_MAX);
 
     lrf.renderLine(p1, p2, ranges);
-    ASSERT_EQ(ranges[0], RANGE_MAX);
-    ASSERT_LT(ranges[N_BEAMS-1], RANGE_MAX);
-    ASSERT_EQ(ranges[lrf.getAngleUpperIndex(p1.x, p1.y)-1], RANGE_MAX);
-    ASSERT_LT(ranges[lrf.getAngleUpperIndex(p1.x, p1.y)], RANGE_MAX);
 
+    uint upper_index_a1 = lrf.getAngleUpperIndex(a1);
+    // the rendered line passes behind the robot. But not through the blindspot. Therefore the first index should remain untouched
+    for (uint i = 0; i< upper_index_a1-1; i++)
+    {
+        ASSERT_EQ(ranges[i], RANGE_MAX) << "range at index [" << i << "] should not be rendered. Instead is rendered to " << ranges[i];
+    }
+    // At a1 we should see the switch between rendered and not rendered. indices above a1 should be rendered. Because points lie on a unit circle they should be <= 1
+    for (uint i = upper_index_a1; i< N_BEAMS; i++)
+    {
+        ASSERT_LE(ranges[i], 1.0) << "range at index [" << i << "] should be rendered to <=1. Instead its value is " << ranges[i];
+    }
```

Comment:
```suggestion
    for (uint i = 0; i<upper_index_a1-1; ++i)
    {
        ASSERT_EQ(ranges[i], RANGE_MAX) << "range at index [" << i << "] should not be rendered. Instead is rendered to " << ranges[i];
    }
    // At a1 we should see the switch between rendered and not rendered. indices above a1 should be rendered. Because points lie on a unit circle they should be <= 1
    for (uint i = upper_index_a1; i<N_BEAMS; ++i)
    {
        ASSERT_LE(ranges[i], 1.0) << "range at index [" << i << "] should be rendered to <=1. Instead its value is " << ranges[i];
    }
```